Registry:  Action history v2 (#1831)
wandb/docs: models/artifacts/explore-and-traverse-an-artifact-graph.mdx

---
description: Traverse direct acyclic W&B Artifact graphs.
title: Explore artifact lineage graphs
---

> W&B tracks the inputs and outputs of runs using directed acyclic graphs (DAGs) called _lineage graphs_. Lineage graphs are visual representations of the relationships between artifacts and runs in an ML experiment. They show how data and models

Reviewer: ```suggestion
W&B tracks the inputs and outputs of runs using directed acyclic graphs (DAGs) called _lineage graphs_. Lineage graphs are visual representations of the relationships between an ML experiment's runs and artifacts. They show how data and models
```
I was having a hard time parsing "and runs", my mind wanted to make it a verb.

flow through different stages of the ML lifecycle, from raw data ingestion to model training and evaluation.

Tracking artifact lineage provides several key advantages:

> * **Reproducibility:** Enables teams to reproduce experiments, models, and results for debugging, experimentation, and validation.

Reviewer: Nit: I think we usually have the `:` outside the bold.

* **Version control:** Tracks changes to artifacts over time, allowing teams to revert to previous data or model versions when needed.
* **Auditing:** Maintains a detailed record of artifacts and transformations to support compliance and governance.

> * **Collaboration:** Improves teamwork by making experiment history transparent, reducing duplicated effort, and accelerating development.

Reviewer: ```suggestion
* **Collaboration:** Helps to improve teamwork by making experiment history transparent, reducing duplicated effort, and accelerating development.
```

> ##  View an artifact's lineage graph

Reviewer: ```suggestion
## View an artifact's lineage graph
```

To view an artifact's lineage graph:

1. Navigate to the W&B App.
2. Select the project that contains the run or artifact you want to explore.
3. Click on the **Artifacts** tab on the left sidebar.
4. Select the **Lineage** tab.

<Frame>
    <img src="/images/artifacts/lineage1.gif" alt="Getting to the Lineage tab"  />
</Frame>

## Enable lineage graph tracking 

> To enable lineage graph tracking, you need to mark artifacts as [inputs](/models/artifacts/explore-and-traverse-an-artifact-graph) or

Reviewer: Are we going away from "turn on"?

[outputs](/models/artifacts/explore-and-traverse-an-artifact-graph#track-the-output-of-a-run) of a run using the W&B Python SDK.

> ###  Track the input of a run

Reviewer: ```suggestion
### Track the input of a run
```

> Mark an artifact as the input (or dependency) of a run with the [`wandb.Run.use_artifact()`](/ref/python/experiments/run/#method-runuse_artifact)

Reviewer: It's a shame to see all this hard wrapping

    <img src="/images/artifacts/lineage3a.gif" alt="Previewing a run"  />
</Frame>

> ##  Artifact clusters

Reviewer: ```suggestion
## Artifact clusters
```

    <img src="/images/artifacts/lineage3b.gif" alt="Searching a run cluster"  />
</Frame>

> ## Programmatically navigate an artifact graph

Reviewer: ```suggestion
## Navigate an artifact graph programmatically
```
Optional feedback to fix a split infinitive
